src/mem/cache/prefetch/despacito_stream.cc

```
@@ -29,45 +99,139 @@ DespacitoStreamPrefetcher::updateSampler(const PrefetchInfo &pfi)
+    stats.nlSampleUpdateReqNotHitTimes++; // If not present, it means no hit
     }
 
     if (timestamp % sampleRate == 0) {
+    stats.nlSampleInsertTimes++;// Analyze insert times
         SamplerEntry *evict_sampler_entry = sampler.findVictim(block_index);
-        updatePatternTable(evict_sampler_entry);
+    stats.nlSampleVictimTouchedTrueTimes += evict_sampler_entry->touched ? 1 : 0; // Analyze victim touched times
+    stats.nlSampleVictimTouchedFalseTimes += evict_sampler_entry->touched ? 0 : 1;
+        // create local snapshots to capture victim/insert data for aggregated record
+        SamplerSnapshot victimSnap;
+        bool haveVictim = false;
```

Cppcheck (2.19.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)